### PR TITLE
Unit testing6

### PR DESCRIPTION
**Pull request by christophercolumbusdog**
Integrate unit testing and refactors

### CODE REVIEW

**Review by mrrush2 — APPROVED** [review]

File: app/src/main/java/com/example/mac/chartr/LocationNameProvider.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package com.example.mac.chartr;
```
Comment (wasabi-muffin) [inline app/src/main/java/com/example/mac/chartr/LocationNameProvider.java:1]: We don't need this file anymore

File: app/src/main/java/com/example/mac/chartr/fragments/SearchFragment.java [hunk 2]
```diff
@@ -18,6 +18,7 @@
 import com.example.mac.chartr.ApiClient;
 import com.example.mac.chartr.ApiInterface;
 import com.example.mac.chartr.CommonDependencyProvider;
+import com.example.mac.chartr.LocationNameProvider;
```
Comment (wasabi-muffin) [inline app/src/main/java/com/example/mac/chartr/fragments/SearchFragment.java:21]: Don't need this

File: app/src/test/java/com/example/mac/chartr/activities/PostTripActivityTest.java [hunk 3]
```diff
@@ -39,6 +33,23 @@ public void setup() {
     }
 
 
+    @Test
+    public void testInitPickers() {
+        PostTripActivity activity = Robolectric.buildActivity(PostTripActivity.class)
+                .create().get();
+
+
+        EditText departureDate = activity.findViewById(R.id.editTextDepartureDate);
```
Comment (wasabi-muffin) [inline app/src/test/java/com/example/mac/chartr/activities/PostTripActivityTest.java:42]: Make these global variables, we only want to findViewById for each object once per file if possible

**Review by wasabi-muffin — CHANGES_REQUESTED** [review]
The LocationProvider is moved to the Trips model, so might need to refactor some of that code/tests

Move your objects outside of functions so you don't have to call findviewbyid multiple times for the same objects

**Review by wasabi-muffin — APPROVED** [review]

**Pull request merged**